controllers/shipwrightbuild_controller_test.go

```diff
@@ -353,13 +353,14 @@ func testShipwrightBuildReconcilerReconcile(t *testing.T, targetNamespace string
 	// Build Controller is created accordingly
 	t.Run("rollout-manifests", func(t *testing.T) {
 		ctx := context.TODO()
-
 		res, err := r.Reconcile(ctx, req)
 		g.Expect(err).To(o.BeNil())
 		g.Expect(res.Requeue).To(o.BeFalse())
-
 		err = c.Get(ctx, deploymentName, &appsv1.Deployment{})
 		g.Expect(err).To(o.BeNil())
+		err = c.Get(ctx, namespacedName, b)
+		g.Expect(err).To(o.BeNil())
+		g.Expect(b.Status.IsReady()).To(o.BeTrue())
```

Comment:
can we construct a test where is ready is expected to report false and we verify that as well ?

Reply:
there is no testing actually handling the falty situation where the deployment is failing

Reply:
yes I understand there is no current testing .... I'm asking if you could take that on ... i.e. I understand it covers elements beyond you specific change here.... i.e. I'm asking some additional work of you

if you don't feel comfortable with that though, I'm OK with that as well.  We can open a tracking issue to cover faulty deployement unit testing separately

Reply:
@gabemontero so i spent sometime looking ot it and cant find a simple way to create the faulty deployment, I thaught about using a bad manifest file pushing this in an env variables as it is in this env variable KO_DATA_PATH, but setting this env variable in the unit test itself is not working, like it is overrided by the global one pushed by the shell when doing make e2e (export KO_DATA_PATH=$(shell pwd)/cmd/manager/kodata)
So
- if you have an idea on how can I do it, I am looking forward to that @gabemontero 
otherwise we can create an issue to track it and add thoses tests later

Reply:
ok  thanks for taking a first pass @jkhelil 

I've opened https://github.com/shipwright-io/operator/issues/22 for now
I included some high level spots to start researching means for the testing, but have not pursued actually investigating what is the best approach from those sources for achieving this

but I'd like to hear if @otaviof and @adambkaplan (the current assignees for this PR) are fine with deferring error testing for your feature until a subsequent PR

Reply:
I think it's worth pursing this in a subsequent pull-request. Until now we assume, and we test it as well, the Build Controller deployment will work, but this operator does not verify the status of the Build Controller is successful.

It indeed goes beyond the scope of this story, but it's a interesting feature to pursue nevertheless! I think we should use #22 to discuss and see how we would like to tackle this.